xla/hlo/transforms/collectives/collective_quantizer.cc

```diff
@@ -307,29 +315,32 @@ absl::Status MatchDequantization(HloInstruction* instr, bool* changed) {
   TF_RETURN_IF_ERROR(
       instr->ReplaceAllUsesWith(subgraph->binary ? new_binary : new_convert));
 
-  *changed = true;
-  VLOG(5) << "Quantized collective " << new_collective->ToShortString();
+  VLOG(5) << "Collective " << instr->ToString() << " has been replaced with "
+          << new_collective->ToString();
 
-  return absl::OkStatus();
+  return true;
 }
 
-absl::Status MatchQuantization(HloInstruction* instr, bool* changed) {
+absl::StatusOr<bool> MatchQuantization(HloInstruction* instr) {
   std::optional<ConversionSubgraph> subgraph;
```

Comment:
You 'd better add some VLOG() to easily tell users/developers why something was labeled as FP8 type but not get quantized or dequantized.

You can copy over the same logic in my PR: https://github.com/openxla/xla/pull/18330

Otherwise when things go wrong, I need to go ahead to modify the binary for adding some necessary VLOG to understand why F8 conversion not happening.